src/swaps/liquid.rs

```diff
@@ -404,37 +401,19 @@ impl LBtcSwapScript {
     }
 
     /// Fetch utxo for script from Electrum
-    pub fn fetch_utxo(&self, network_config: &ElectrumConfig) -> Result<(OutPoint, TxOut), Error> {
-        let electrum_client = network_config.clone().build_client()?;
+    pub async fn fetch_utxo<LC: LiquidClient, N: LiquidNetworkConfig<LC>>(
+        &self,
+        network_config: &N,
```

Comment:
Passing a config forces this function to call `build_client` (and create a new client) which prevents from the caller to reuse a client implementation. In electrum we have seen it has a big performance penalty.
I know this was the implementation before this PR but perhaps if we are going for such a big change we can improve it?
I commented on this method as an example but this pattern repeats itself.

Reply:
Thanks @roeierez. I agree. I've made the methods take a client instead of the config in [536cf8d](https://github.com/SatoshiPortal/boltz-rust/pull/91/commits/536cf8d3dcb342560e0cb2f7d5686a7fe3a48d9a).